implementation/src/main/java/io/smallrye/config/mapper/ConfigMapping.java

```diff
@@ -602,10 +600,8 @@ public static Builder builder() {
         return new Builder();
     }
 
-    public <B extends SmallRyeConfigBuilder> B registerDefaultValues(B builder) {
-        Assert.checkNotNullParam("builder", builder);
-        builder.withSources(new DefaultValuesConfigSource(defaultValues));
-        return builder;
+    public Map<String, String> getDefaultValues() {
+        return Collections.unmodifiableMap(defaultValues.toMap());
```

Comment:
This is incorrect.  The default values cannot be represented by a flat map because they are pattern-backed.  For example a map value of `foo.*.bar` could have a default value of `baz`, and so `baz` would have to be returned for any *matching* key.

Reply:
We can back this up with the `KeyMap` work, but I did implement the Default API separately first, to not depend yet on the Mapping.